homeassistant/components/bryant_evolution/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config flow for Bryant Evolution integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from evolutionhttp import BryantEvolutionLocalClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_FILENAME
+from homeassistant.helpers.typing import UNDEFINED
+
+from .const import CONF_SYSTEM_ZONE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_FILENAME, default="/dev/ttyUSB0"): str,
+    }
+)
+
+
+async def _enumerate_sz(tty: str) -> list[tuple[int, int]]:
+    """Return (system, zone) tuples for each system+zone accessible through tty."""
+    return [
+        (1, x.zone_id) for x in await BryantEvolutionLocalClient.enumerate_zones(1, tty)
+    ] + [
+        (2, x.zone_id) for x in await BryantEvolutionLocalClient.enumerate_zones(2, tty)
+    ]
```

Comment:
```suggestion
    return [
        (system_id, zone.zone_id)
        for zone in await BryantEvolutionLocalClient.enumerate_zones(system_id, tty)
        for system_id in (1, 2)
    ]
```
Just an experiment, I think this would achieve the same

Reply:
system_id ends up as not defined on the second line

Reply:
sorry, third -- this one: for zone in await BryantEvolutionLocalClient.enumerate_zones(system_id, tty)

(I fixed (system_id, zone.zone_id) --> (zone.system_id, zone.zone_id))